### DISCUSSION TITLE
How to give an error upon signup if email address is already in use?

### DISCUSSION THREAD

**Discussion by zivtamary** [supabase/supabase#10817] [Questions]
I've seen some threads on this topic in the past but no answers were given, but only hope that it'll get fixed.

When I use supabase.signUp and that user already exists in the system, it just returns an object with that user, and saying to check that email for verification. How should I approach this? I thought maybe I'd create simultaneously a new user in another table, and check there if that email exists. but it sounds inefficient

Has anyone done this before?

Thanks

**Comment by jgreywolf** [comment]
Old question - but see the following:

https://github.com/orgs/supabase/discussions/1282

>if signUp() is called for an existing confirmed user:
If Confirm email is enabled in [your project](https://supabase.com/dashboard/project/_/auth/providers), an obfuscated/fake user object is returned.
If Confirm email is disabled, the error message, User already registered is returned.
To fetch the currently logged-in user, refer to [getUser()](https://supabase.com/docs/reference/javascript/auth-getuser).
